src/common/components/vanilla-modules/forms/form-group-input.js

```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+
+import styles from '../../../style/vanilla/css/forms.css';
+
+export default function FormGroupInput(props) {
+  const { disabled, name, label, type, placeholder } = props;
+  const id = `ID_${name}`;
+  const status = props.touched ? ( props.valid ? 'p-form-validation is-success' : 'p-form-validation is-error') : null;
+
+  return (
+    <div className={ `${styles['p-form__group']} ${styles[status]}` }>
+      <label htmlFor={ id } className={ styles['p-form__label'] }>
+        { label }
+      </label>
+      <div className={ styles['p-form__control'] }>
+        <input
+          id={ id }
+          name={ props.sensitive ? null : name }
+          data-name={ name }
+          type={ type }
+          required={ props.required }
+          disabled={ disabled }
+          placeholder={ placeholder }
+          onChange={ props.onChange }
+          onBlur={ props.onBlur }
+          value={ props.value || '' }
```

Comment:
You marked `value` as required property, but you check if it has any value. There are couple of issues with that.

One - you cannot have `value=0`, because 0 will be false and it will end up as empty string.

But more importantly if you require value property and bind input value this property it cannot be changed by user unless there an onChange handler that updates the value.
As a result you can't type anything to such input (it's practically read only).

Binding input values is a good pattern in general, but may make it difficult to make simple forms that don't need to be bound to application state.

More on that if you're interested: https://facebook.github.io/react/docs/forms.html#controlled-components

But shortly speaking, ideally value should not be required to make it possible to create so called 'uncontrolled' input that user can type in without any onChange handler making it happen.